### ISSUE TITLE
Tokenizer issue related to comment character in strings

### ISSUE THREAD

**Issue by paulholden2** [project-eutopia/keisan#101]
Hello! I ran into another issue with the tokenizer in (related to #94) regards to strings whereby having a hash character (which marks the remainder of the line as a comment) in a string literal will cause tokenization to fail, as the closing quote at the end of the line is regarded as part of the comment.

Version: 0.8.3

Executing the following demonstrates the issue:

```rb
Keisan::Calculator.new.ast('"#1"')
# raises Keisan::Exceptions::TokenizingError (Tokenizing error, no closing quote ')

Keisan::Calculator.new.ast("'#1'")
# raises Keisan::Exceptions::TokenizingError (Tokenizing error, no closing quote ")
```

Expected behavior: comment sequences should be treated as part of the string literal, only being ignored if encountered outside a string literal. I'm guessing the expression syntax is supposed to be much like actual Ruby, and I know that string literals like '#' definitely treat the hash as part of the literal and not a comment sequence.

**Comment by paulholden2** [comment]
Just realized my examples are switched. First one reports no closing quote ", second one is '.

**Comment by project-eutopia** [comment]
Thanks for catching this! Did you want to checkout the `fix-comments-and-strings` branch in #102 to see if this fixes things before I submit it and push a new version? Here are some samples for what things look like with this change:

```
➜  keisan git:(fix-comments-and-strings) bin/console
[1] pry(main)> Keisan::Calculator.new.ast("'#1'")
=> #<Keisan::AST::String:0x0000557a6e2010f0 @content="#1">
[2] pry(main)> Keisan::Calculator.new.ast("# '#1''\nx + '#foo'#bar")
=> #<Keisan::AST::Plus:0x0000557a6e30cee0 @children=[#<Keisan::AST::Variable:0x0000557a6e30d930 @name="x">, #<Keisan::AST::String:0x0000557a6e30d340 @content="#foo">], @parsing_operators=[#<Keisan::Parsing::Plus:0x0000557a6e30e5b0>]>
```

Notice how in the second case the first line which is just a comment is stripped out, and the trailing comment in the second line is stripped out correctly.  All that is left is the `x + '#foo'` part which makes up the AST.

**Comment by paulholden2** [comment]
Works perfectly! My use doesn't require the use of comments but it's awesome to see it working in your 2nd example. Your prompt fixes are amazing by the way!

**Comment by project-eutopia** [comment]
Glad to help!  Your submitted bugs have contributed to making this a better library.

I just pushed the fix, and updated the gem to version 0.8.4 (https://rubygems.org/gems/keisan/versions/0.8.4).  I installed the updated version locally, and confirmed that the changes work in my terminal:

```
➜  keisan git:(master) gem install keisan
Successfully installed keisan-0.8.4
Parsing documentation for keisan-0.8.4
Installing ri documentation for keisan-0.8.4
Done installing documentation for keisan after 0 seconds
1 gem installed
➜  keisan git:(master) irb
2.6.3 :001 > require "keisan"
 => true
2.6.3 :002 > Keisan::Calculator.new.evaluate("# '#'\"\n1+2#ignore")
 => 3
2.6.3 :003 > Keisan::Calculator.new.ast('"#1"')
 => #<Keisan::AST::String:0x00005567ecddd778 @content="#1">
```
